### PR TITLE
Cumulus 1160 add integration tests

### PR DESCRIPTION
**Pull request by Jkovarik**
**Summary:** Summary of changes

Adds integration scripts to bamboo script set.  See ticket for AC specifics.  

Addresses [CUMULUS-1160](https://bugs.earthdata.nasa.gov/browse/CUMULUS-1160)

### CODE REVIEW

File: example/scripts/tests-parallel.sh [hunk 1]
```diff
@@ -9,16 +9,13 @@ echo Running parallel integration tests
...
 TESTS=$(find spec/parallel -type f -name '*spec.js' -or -name '*Spec.js')
-
 testOutputDir=scripts/test_output
 
 rm -r -f $testOutputDir
 mkdir -p $testOutputDir
 
-./node_modules/.bin/parallel -j 4 sh scripts/run_test.sh  $testOutputDir ::: $TESTS
-
+echo "" | ./node_modules/.bin/parallel -j 6 sh scripts/run_test.sh  $testOutputDir ::: $TESTS
```
Comment (Jkovarik) [inline example/scripts/tests-parallel.sh]: This is an annoyance in parallel - without a terminal/pseudoterminal the process never ends.

File: example/scripts/lock-stack.js [hunk 2]
```diff
@@ -1,5 +1,4 @@
 /* eslint no-console: "off" */
```
Comment (Jkovarik) [inline example/scripts/lock-stack.js]: The changes here are part of https://github.com/nasa/cumulus/pull/975

File: bamboo/bootstrap-integration-tests.sh [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  cd example
+  set +e
+
+  # Wait for the stack to be available
+  node ./scripts/lock-stack.js true $DEPLOYMENT
+  LOCK_EXISTS_STATUS=$?
+  echo "Locking status $LOCK_EXISTS_STATUS"
+
+  COUNTER=0;
+  while [[ $LOCK_EXISTS_STATUS = 100 ]]; do
```
Comment (Jkovarik) [inline bamboo/bootstrap-integration-tests.sh]: This logic maps to the fix in https://github.com/nasa/cumulus/pull/975

File: bamboo/select-stack.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  const branch = process.env.BRANCH
+
+  if (!branch) return cb('none');
+
+  // Nightly cron job
+  //if (process.env.TRAVIS_EVENT_TYPE === 'cron') return cb('cumulus-nightly');
+
+  if (branch === 'master') return cb('cumulus-from-source');
+
+  const stacks = {
```
Comment (laurenfrederick) [inline bamboo/select-stack.js]: Have we verified on Bamboo that the author comes through this way and that these will work?

File: example/scripts/tests-parallel.sh (hunk 1 above)
Comment (laurenfrederick) [reply example/scripts/tests-parallel.sh]: Maybe put a comment in there.

**Comment by Jkovarik** [comment]
Need to bring in changes from #975 before merge

**Pull request merged**